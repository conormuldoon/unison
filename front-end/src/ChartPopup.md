ChartPopup example:

```js
import {useState} from 'react';
const [display,setDisplay] = useState(false);
function toggleDisplay(){
    if(display)setDisplay(false);
    else setDisplay(true);
}

if(display){
    window.scrollTo(0,0);
}

<div>
    
    <button onClick={toggleDisplay}>{(display)?"Hide":"Display"}</button>
    {display&&<ChartPopup closePopup={toggleDisplay} varCur="Precipitation" location="UCD" fromDate="4/10/2019" toDate="5/10/2019" />}
</div>
```