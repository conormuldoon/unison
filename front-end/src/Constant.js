
export const FORMAT = "D/M/YYYY";

export const API = "/apiacc";

export const LPER = "20\u1d57\u02b0 percentile";
export const HPER = "80\u1d57\u02b0 percentile";
export const MEDIAN = "Median";

export const LOW = "Low";
export const MEDIUM = "Medium";
export const HIGH = "High";
export const PRECIP = "Precipitation";
export const CL = "Cloud Level";
//export const API = "";

// Map coordinates
//export const CENTRE_LAT = 53.35014;
//export const CENTRE_LON = -6.266155;

// Oslo
export const CENTRE_LAT = 59.922326;
export const CENTRE_LON = 10.751560;
