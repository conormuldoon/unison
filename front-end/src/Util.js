
import {FORMAT} from './Constant';

import  {
  formatDate
} from 'react-day-picker/moment';

export const fromDate = () =>{

  const today=new Date(Date.now());
  const fDate=new Date(Date.now());
  fDate.setMonth(today.getMonth()-1);

  return formatDate(fDate,FORMAT);
}

export const today = () => {
  return formatDate(new Date(Date.now()), FORMAT);
};

const postObject = (body) =>{
  return {
    method: 'POST',
    headers: new Headers({
               'Content-Type': 'application/x-www-form-urlencoded',
      }),
    body: body
  }
}

const locCred = (location,uname,pword) =>{
  return 'location='+location+'&username='+uname+'&password='+pword;
}

export const removePostObject = (location,uname,pword) =>{
  return postObject(locCred(location,uname,pword));

}

export const locationPostObject = (location,uname,pword,lon,lat) =>{
  return postObject(locCred(location,uname,pword)+'&longitude='+lon+'&latitude='+lat);
}

export const varMapping = (varCur) =>{
  let variRequest=varCur.replace(/ /g,'');
  variRequest=variRequest.charAt(0).toLowerCase() + variRequest.slice(1);
  return variRequest;
}
