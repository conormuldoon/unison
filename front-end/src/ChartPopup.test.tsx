import { enableFetchMocks } from 'jest-fetch-mock';
enableFetchMocks();

import fetchMock from 'jest-fetch-mock';
import React from 'react';
import { render, screen } from "@testing-library/react";
import ChartPopup from './ChartPopup';
import { PRECIP, CL } from './Constant';
import { chartText } from './Util';

const fromDate = '1-2-2018';
const toDate = '7-10-2019';

const location = {
  "name": "UCD",
  "_links": {
    "self": {
      "href": "http://localhost:8080/locationCollection/UCD"
    },
    "cloudiness": {
      "href": "http://localhost:8080/locationCollection/UCD/cloudiness{?fromDate,toDate}",
      "templated": true
    },
    "cloudLevel": {
      "href": "http://localhost:8080/locationCollection/UCD/cloudLevel{?fromDate,toDate}",
      "templated": true
    },
    "dewPoint": {
      "href": "http://localhost:8080/locationCollection/UCD/dewPoint{?fromDate,toDate}",
      "templated": true
    },
    "humidity": {
      "href": "http://localhost:8080/locationCollection/UCD/humidity{?fromDate,toDate}",
      "templated": true
    },
    "precipitation": {
      "href": "http://localhost:8080/locationCollection/UCD/precipitation{?fromDate,toDate}",
      "templated": true
    },
    "pressure": {
      "href": "http://localhost:8080/locationCollection/UCD/pressure{?fromDate,toDate}",
      "templated": true
    },
    "temperature": {
      "href": "http://localhost:8080/locationCollection/UCD/temperature{?fromDate,toDate}",
      "templated": true
    },
    "windDirection": {
      "href": "http://localhost:8080/locationCollection/UCD/windDirection{?fromDate,toDate}",
      "templated": true
    },
    "windSpeed": {
      "href": "http://localhost:8080/locationCollection/UCD/windSpeed{?fromDate,toDate}",
      "templated": true
    }
  }
}


const uri = "http://localhost:8080/locationCollection/UCD/precipitation?fromDate=" + fromDate + "&toDate=" + toDate;

const chartPopup = <ChartPopup curVar='Precipitation'
  closePopup={() => { }} uri={uri} name={location.name} />;

const apiRequest = 'end:?fromDate=1-2-2018&toDate=7-10-2019';


const data = [{ "date": "2019-02-23T19:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-23T20:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-23T21:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-23T22:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-23T23:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-26T18:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T12:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T13:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T14:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T15:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T16:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T17:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T18:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T19:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T20:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T21:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T22:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T23:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-26T00:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-26T01:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-26T02:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-26T03:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-26T04:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-26T05:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-26T06:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-26T07:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-26T08:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-26T09:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-26T10:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-26T11:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-26T12:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-26T13:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-26T14:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-26T15:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-26T16:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-26T17:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-24T00:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-24T01:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-24T02:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-24T03:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-24T04:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-24T05:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-24T06:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-24T07:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-24T08:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.1 } }, { "date": "2019-02-24T09:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.1 } }, { "date": "2019-02-24T10:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-24T11:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-24T12:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-24T13:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-24T14:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-24T15:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-24T16:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-24T17:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-24T18:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-24T19:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-24T20:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-24T21:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-24T22:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-24T23:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T00:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T01:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T02:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T03:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T04:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T05:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T06:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T07:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T08:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T09:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T10:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-25T11:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T00:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T01:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T02:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T03:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T04:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T05:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T06:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T07:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T08:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T09:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T10:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T11:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T12:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T13:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T14:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T15:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T16:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T17:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T18:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T19:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T20:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T21:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T22:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-27T23:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-28T00:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-28T01:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-28T02:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-28T03:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-28T04:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-28T05:00:00.000+0000", "precipitation": { "value": 0.2, "minvalue": 0.1, "maxvalue": 0.3 } }, { "date": "2019-02-26T19:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-26T20:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-26T21:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-26T22:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }, { "date": "2019-02-26T23:00:00.000+0000", "precipitation": { "value": 0.0, "minvalue": 0.0, "maxvalue": 0.0 } }];

it('renders without crashing', async () => {


  fetchMock.mockOnce("[]");

  render(chartPopup);
  await screen.findByRole('button', { name: 'email' });

});



it('mathes snapshot', async () => {

  fetchMock.mockResponse(JSON.stringify(data));

  const { container } = render(chartPopup);
  await screen.findByText('Precipitation data from ' + location.name);
  expect(container).toMatchSnapshot();

  fetchMock.resetMocks();

});


it('displays text for the selected variable and location', async () => {


  const VAR_OPT = [PRECIP, 'Humidity', 'Wind Direction', 'Wind Speed', 'Cloudiness', CL, 'Dew Point', 'Pressure', 'Temperature'];
  for (const vo of VAR_OPT) {
    let sOpt = chartText(vo);
    for (const loc of [{
      "name": "Dublin",
      "_links": {
        "self": {
          "href": "http://localhost:8080/locationCollection/Dublin"
        },
        "cloudiness": {
          "href": "http://localhost:8080/locationCollection/Dublin/cloudiness{?fromDate,toDate}",
          "templated": true
        },
        "cloudLevel": {
          "href": "http://localhost:8080/locationCollection/Dublin/cloudLevel{?fromDate,toDate}",
          "templated": true
        },
        "dewPoint": {
          "href": "http://localhost:8080/locationCollection/Dublin/dewPoint{?fromDate,toDate}",
          "templated": true
        },
        "humidity": {
          "href": "http://localhost:8080/locationCollection/Dublin/humidity{?fromDate,toDate}",
          "templated": true
        },
        "precipitation": {
          "href": "http://localhost:8080/locationCollection/Dublin/precipitation{?fromDate,toDate}",
          "templated": true
        },
        "pressure": {
          "href": "http://localhost:8080/locationCollection/Dublin/pressure{?fromDate,toDate}",
          "templated": true
        },
        "temperature": {
          "href": "http://localhost:8080/locationCollection/Dublin/temperature{?fromDate,toDate}",
          "templated": true
        },
        "windDirection": {
          "href": "http://localhost:8080/locationCollection/Dublin/windDirection{?fromDate,toDate}",
          "templated": true
        },
        "windSpeed": {
          "href": "http://localhost:8080/locationCollection/Dublin/windSpeed{?fromDate,toDate}",
          "templated": true
        }
      }
    }, {
      "name": "London",
      "_links": {
        "self": {
          "href": "http://localhost:8080/locationCollection/London"
        },
        "cloudiness": {
          "href": "http://localhost:8080/locationCollection/London/cloudiness{?fromDate,toDate}",
          "templated": true
        },
        "cloudLevel": {
          "href": "http://localhost:8080/locationCollection/London/cloudLevel{?fromDate,toDate}",
          "templated": true
        },
        "dewPoint": {
          "href": "http://localhost:8080/locationCollection/London/dewPoint{?fromDate,toDate}",
          "templated": true
        },
        "humidity": {
          "href": "http://localhost:8080/locationCollection/London/humidity{?fromDate,toDate}",
          "templated": true
        },
        "precipitation": {
          "href": "http://localhost:8080/locationCollection/London/precipitation{?fromDate,toDate}",
          "templated": true
        },
        "pressure": {
          "href": "http://localhost:8080/locationCollection/London/pressure{?fromDate,toDate}",
          "templated": true
        },
        "temperature": {
          "href": "http://localhost:8080/locationCollection/London/temperature{?fromDate,toDate}",
          "templated": true
        },
        "windDirection": {
          "href": "http://localhost:8080/locationCollection/London/windDirection{?fromDate,toDate}",
          "templated": true
        },
        "windSpeed": {
          "href": "http://localhost:8080/locationCollection/London/windSpeed{?fromDate,toDate}",
          "templated": true
        }
      }
    }]) {

      fetchMock.mockOnce(JSON.stringify(loc));

      const uri = "http://localhost:8080/locationCollection/London/windSpeed?fromDate=" + fromDate + "&toDate=" + toDate;
      render(<ChartPopup curVar={vo} name={loc.name} uri={uri} closePopup={() => { }} />);

      await screen.findByText(sOpt + ' data from ' + loc.name);
    }
  }

  fetchMock.resetMocks();
});



it('displays a lower case letter for second word', async () => {

  fetchMock.mockOnce("[]");

  const wdURI = "http://localhost:8080/locationCollection/London/windDirection?fromDate=" + fromDate + "&toDate=" + toDate;
  render(<ChartPopup curVar={'Wind Direction'} name={location.name} uri={wdURI}
    closePopup={() => { }} />);

  await screen.findByText('Wind direction data from ' + location.name);



});

it('adds a chart', async () => {

  fetchMock.mockOnce(JSON.stringify(data));

  render(chartPopup);
  await screen.findByTestId('chart');

});

