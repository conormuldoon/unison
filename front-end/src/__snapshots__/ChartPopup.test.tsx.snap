// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mathes snapshot 1`] = `
<div>
  <div
    data-testid="chart-div"
    id="popupdiv"
    style="text-align: center;"
  >
    <div
      id="iicon"
    >
      <svg
        color="rgb(192, 57, 43)"
        fill="currentColor"
        height="20"
        size="20"
        stroke="currentColor"
        stroke-width="0"
        style="color: rgb(192, 57, 43);"
        viewBox="0 0 512 512"
        width="20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M405 136.798L375.202 107 256 226.202 136.798 107 107 136.798 226.202 256 107 375.202 136.798 405 256 285.798 375.202 405 405 375.202 285.798 256z"
        />
      </svg>
      <button
        aria-label="email"
        class="react-share__ShareButton"
        style="background-color: transparent; padding: 0px; cursor: pointer;"
      >
        <svg
          height="24"
          viewBox="0 0 64 64"
          width="24"
        >
          <rect
            fill="#7f7f7f"
            height="64"
            rx="0"
            ry="0"
            width="64"
          />
          <path
            d="M17,22v20h30V22H17z M41.1,25L32,32.1L22.9,25H41.1z M20,39V26.6l12,9.3l12-9.3V39H20z"
            fill="white"
          />
        </svg>
      </button>
      <button
        aria-label="whatsapp"
        class="react-share__ShareButton"
        style="background-color: transparent; padding: 0px; cursor: pointer;"
      >
        <svg
          height="24"
          viewBox="0 0 64 64"
          width="24"
        >
          <rect
            fill="#25D366"
            height="64"
            rx="0"
            ry="0"
            width="64"
          />
          <path
            d="m42.32286,33.93287c-0.5178,-0.2589 -3.04726,-1.49644 -3.52105,-1.66732c-0.4712,-0.17346 -0.81554,-0.2589 -1.15987,0.2589c-0.34175,0.51004 -1.33075,1.66474 -1.63108,2.00648c-0.30032,0.33658 -0.60064,0.36247 -1.11327,0.12945c-0.5178,-0.2589 -2.17994,-0.80259 -4.14759,-2.56312c-1.53269,-1.37217 -2.56312,-3.05503 -2.86603,-3.57283c-0.30033,-0.5178 -0.03366,-0.80259 0.22524,-1.06149c0.23301,-0.23301 0.5178,-0.59547 0.7767,-0.90616c0.25372,-0.31068 0.33657,-0.5178 0.51262,-0.85437c0.17088,-0.36246 0.08544,-0.64725 -0.04402,-0.90615c-0.12945,-0.2589 -1.15987,-2.79613 -1.58964,-3.80584c-0.41424,-1.00971 -0.84142,-0.88027 -1.15987,-0.88027c-0.29773,-0.02588 -0.64208,-0.02588 -0.98382,-0.02588c-0.34693,0 -0.90616,0.12945 -1.37736,0.62136c-0.4712,0.5178 -1.80194,1.76053 -1.80194,4.27186c0,2.51134 1.84596,4.945 2.10227,5.30747c0.2589,0.33657 3.63497,5.51458 8.80262,7.74113c1.23237,0.5178 2.1903,0.82848 2.94111,1.08738c1.23237,0.38836 2.35599,0.33657 3.24402,0.20712c0.99159,-0.15534 3.04985,-1.24272 3.47963,-2.45956c0.44013,-1.21683 0.44013,-2.22654 0.31068,-2.45955c-0.12945,-0.23301 -0.46601,-0.36247 -0.98382,-0.59548m-9.40068,12.84407l-0.02589,0c-3.05503,0 -6.08417,-0.82849 -8.72495,-2.38189l-0.62136,-0.37023l-6.47252,1.68286l1.73463,-6.29129l-0.41424,-0.64725c-1.70875,-2.71846 -2.6149,-5.85116 -2.6149,-9.07706c0,-9.39809 7.68934,-17.06155 17.15993,-17.06155c4.58253,0 8.88029,1.78642 12.11655,5.02268c3.23625,3.21036 5.02267,7.50812 5.02267,12.06476c-0.0078,9.3981 -7.69712,17.06155 -17.14699,17.06155m14.58906,-31.58846c-3.93529,-3.80584 -9.1133,-5.95471 -14.62789,-5.95471c-11.36055,0 -20.60848,9.2065 -20.61625,20.52564c0,3.61684 0.94757,7.14565 2.75211,10.26282l-2.92557,10.63564l10.93337,-2.85309c3.0136,1.63108 6.4052,2.4958 9.85634,2.49839l0.01037,0c11.36574,0 20.61884,-9.2091 20.62403,-20.53082c0,-5.48093 -2.14111,-10.64081 -6.03239,-14.51915"
            fill="white"
          />
        </svg>
      </button>
      <button
        aria-label="telegram"
        class="react-share__ShareButton"
        style="background-color: transparent; padding: 0px; cursor: pointer;"
      >
        <svg
          height="24"
          viewBox="0 0 64 64"
          width="24"
        >
          <rect
            fill="#37aee2"
            height="64"
            rx="0"
            ry="0"
            width="64"
          />
          <path
            d="m45.90873,15.44335c-0.6901,-0.0281 -1.37668,0.14048 -1.96142,0.41265c-0.84989,0.32661 -8.63939,3.33986 -16.5237,6.39174c-3.9685,1.53296 -7.93349,3.06593 -10.98537,4.24067c-3.05012,1.1765 -5.34694,2.05098 -5.4681,2.09312c-0.80775,0.28096 -1.89996,0.63566 -2.82712,1.72788c-0.23354,0.27218 -0.46884,0.62161 -0.58825,1.10275c-0.11941,0.48114 -0.06673,1.09222 0.16682,1.5716c0.46533,0.96052 1.25376,1.35737 2.18443,1.71383c3.09051,0.99037 6.28638,1.93508 8.93263,2.8236c0.97632,3.44171 1.91401,6.89571 2.84116,10.34268c0.30554,0.69185 0.97105,0.94823 1.65764,0.95525l-0.00351,0.03512c0,0 0.53908,0.05268 1.06412,-0.07375c0.52679,-0.12292 1.18879,-0.42846 1.79109,-0.99212c0.662,-0.62161 2.45836,-2.38812 3.47683,-3.38552l7.6736,5.66477l0.06146,0.03512c0,0 0.84989,0.59703 2.09312,0.68132c0.62161,0.04214 1.4399,-0.07726 2.14229,-0.59176c0.70766,-0.51626 1.1765,-1.34683 1.396,-2.29506c0.65673,-2.86224 5.00979,-23.57745 5.75257,-27.00686l-0.02107,0.08077c0.51977,-1.93157 0.32837,-3.70159 -0.87096,-4.74991c-0.60054,-0.52152 -1.2924,-0.7498 -1.98425,-0.77965l0,0.00176zm-0.2072,3.29069c0.04741,0.0439 0.0439,0.0439 0.00351,0.04741c-0.01229,-0.00351 0.14048,0.2072 -0.15804,1.32576l-0.01229,0.04214l-0.00878,0.03863c-0.75858,3.50668 -5.15554,24.40802 -5.74203,26.96472c-0.08077,0.34417 -0.11414,0.31959 -0.09482,0.29852c-0.1756,-0.02634 -0.50045,-0.16506 -0.52679,-0.1756l-13.13468,-9.70175c4.4988,-4.33199 9.09945,-8.25307 13.744,-12.43229c0.8218,-0.41265 0.68483,-1.68573 -0.29852,-1.70681c-1.04305,0.24584 -1.92279,0.99564 -2.8798,1.47502c-5.49971,3.2626 -11.11882,6.13186 -16.55882,9.49279c-2.792,-0.97105 -5.57873,-1.77704 -8.15298,-2.57601c2.2336,-0.89555 4.00889,-1.55579 5.75608,-2.23009c3.05188,-1.1765 7.01687,-2.7042 10.98537,-4.24067c7.94051,-3.06944 15.92667,-6.16346 16.62028,-6.43037l0.05619,-0.02283l0.05268,-0.02283c0.19316,-0.0878 0.30378,-0.09658 0.35471,-0.10009c0,0 -0.01756,-0.05795 -0.00351,-0.04566l-0.00176,0zm-20.91715,22.0638l2.16687,1.60145c-0.93418,0.91311 -1.81743,1.77353 -2.45485,2.38812l0.28798,-3.98957"
            fill="white"
          />
        </svg>
      </button>
    </div>
    <div
      style="margin-top: 10px; margin-bottom: 20px; text-align: center;"
    >
      Precipitation data from UCD
    </div>
    <div
      id="chartdiv"
    >
      <div
        data-testid="chart"
        id="singlevar"
      />
    </div>
  </div>
</div>
`;
